game-core/src/main/java/games/strategy/engine/framework/lookandfeel/LookAndFeel.java

```diff
@@ -61,43 +49,63 @@
-        SubstanceDustCoffeeLookAndFeel.class.getName(), SubstanceDustLookAndFeel.class.getName(),
-        SubstanceGeminiLookAndFeel.class.getName(), SubstanceGraphiteAquaLookAndFeel.class.getName(),
-        SubstanceGraphiteChalkLookAndFeel.class.getName(), SubstanceGraphiteGlassLookAndFeel.class.getName(),
-        SubstanceGraphiteGoldLookAndFeel.class.getName(), SubstanceGraphiteLookAndFeel.class.getName(),
-        SubstanceMagellanLookAndFeel.class.getName(), SubstanceMarinerLookAndFeel.class.getName(),
-        SubstanceMistAquaLookAndFeel.class.getName(), SubstanceMistSilverLookAndFeel.class.getName(),
-        SubstanceModerateLookAndFeel.class.getName(), SubstanceNebulaBrickWallLookAndFeel.class.getName(),
-        SubstanceNebulaLookAndFeel.class.getName(), SubstanceOfficeBlack2007LookAndFeel.class.getName(),
-        SubstanceOfficeBlue2007LookAndFeel.class.getName(), SubstanceOfficeSilver2007LookAndFeel.class.getName(),
-        SubstanceRavenLookAndFeel.class.getName(), SubstanceSaharaLookAndFeel.class.getName(),
-        SubstanceTwilightLookAndFeel.class.getName()));
-    return substanceLooks;
+    final List<String> lookAndFeelClassNames = new ArrayList<>();
+    lookAndFeelClassNames.addAll(getInstalledLookAndFeelClassNames());
+    lookAndFeelClassNames.addAll(getSubstanceLookAndFeelClassNames());
+    return lookAndFeelClassNames;
   }
 
-  /**
-   * Sets the user's preferred Look-And-Feel. If not available, the system Look-And-Feel will be used.
-   *
-   * @throws IllegalStateException If this method is not called from the EDT.
-   */
-  public static void setupLookAndFeel() {
-    setupLookAndFeel(ClientSetting.LOOK_AND_FEEL_PREF.value());
+  private static Collection<String> getInstalledLookAndFeelClassNames() {
+    return Arrays.stream(UIManager.getInstalledLookAndFeels())
+        .map(UIManager.LookAndFeelInfo::getClassName)
+        .collect(Collectors.toList());
+  }
+
+  private static Collection<String> getSubstanceLookAndFeelClassNames() {
+    return Arrays.asList(
+        substance("Autumn"),
+        substance("BusinessBlackSteel"),
+        substance("BusinessBlueSteel"),
+        substance("Business"),
+        substance("Cerulean"),
+        substance("CremeCoffee"),
+        substance("Creme"),
+        substance("DustCoffee"),
+        substance("Dust"),
+        substance("Gemini"),
+        substance("GraphiteAqua"),
+        substance("GraphiteChalk"),
+        substance("GraphiteGlass"),
+        substance("GraphiteGold"),
+        substance("Graphite"),
+        substance("Magellan"),
+        substance("Mariner"),
+        substance("MistAqua"),
+        substance("MistSilver"),
+        substance("Moderate"),
+        substance("NebulaBrickWall"),
+        substance("Nebula"),
+        substance("OfficeBlack2007"),
+        substance("OfficeBlue2007"),
+        substance("OfficeSilver2007"),
+        substance("Raven"),
+        substance("Sahara"),
+        substance("Twilight"));
```

Comment:
I just want to point out that it would be far less code when simply putting all the strings into the list and then mapping all entries with the method.

Reply:
:+1: 000729679